graylog2-server/src/main/java/org/graylog2/migrations/V20190730000000_CreateDefaultEventsConfiguration.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog2.migrations;
+
+import org.graylog.events.configuration.EventsConfiguration;
+import org.graylog.events.configuration.EventsConfigurationProvider;
+import org.graylog2.plugin.cluster.ClusterConfigService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.time.ZonedDateTime;
+import java.util.Optional;
+
+public class V20190730000000_CreateDefaultEventsConfiguration extends Migration {
```

Comment:
I think we don't need a migration for this. The provider can just return a default instance of the config when there is no entry in the database. The first time a user updates the configuration it will be written to the database.

Reply:
The cluster config resource is not providing any data if there is nothing in the database so the UI component will not load. I don't see at which point we could force the creation through an access?

Reply:
Ah, I see. Okay, then we need the migration. :confused: Good catch!